EIP-5485: Legitimacy, Jurisdiction and Sovereignty (#5485)
ethereum/EIPs: EIPS/eip-5485.md

---
eip: 5485
title: Legitimacy, Jurisdiction and Sovereignty

> description: <Description is one full (short) sentence>

Reviewer: Missing

author: Zainan Victor Zhou (@xinbenlv)
discussions-to: https://ethereum-magicians.org/t/erc-5485-interface-for-legitimacy-jurisdiction-and-sovereignty/10425
status: Draft
type: Standards Track
category: ERC
created: 2022-08-17

> requires: 165

Reviewer: ```suggestion
requires: 165, 5485
```

---

## Abstract
An interface for legitimacy, jurisdiction and sovereignty.

## Motivation
Provide a way for compliant smart contracts to declare their legitimacy lineage, jurisdiction they observe, and sovereignty if they choose to not fall onto any jurisdiction.

For example. If `ContractA` gains legitimacy from `ContractB`, `ContractB` gains legitimacy from `ContractC` but `ContractC` doesn't gain legitimacy from any one, ContractC is considered the "jurisdiction" that both `ContractB` and `ContractA` observe.

> Both `ContractA` and `ContractB` MUST return address of `ContractC` as the result of jurisdiction.

Reviewer: ```suggestion

Both `ContractA` and `ContractB` MUST return address of `ContractC` as the result of jurisdiction.
```

ContractB represents a **The Secretary of State of the country**. The ContractC represents **Supreme Court of the Country**.

Another real example is a contract the declared "self-sovereignty" that doesn't follow any jurisdiction.

This interface supports both cases, providing a way to allow smart contract to determine if they want to allow / prohibit interaction based
on the sovereignty.

For example A country might want to require any digital money service's all smart contracts to observe their [EIP-5485](./eip-5485.md) jurisdiction before they are allow to operate money in their (real world) legal jurisdiction.

On the other hand, a DAO with strong decentralized ideology might choose to only inter-operate with EOA
or "self-sovereigned" smart contracts to avoid being affected by any countries.

## Specification
The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.

1. Compliant contract MUSTS implement the following interface.

```solidity
interface IERC5485 {
    function legitimatedBy() public view returns (address);

>     function jurisdiction() public view returns (address);

Reviewer: Need definitions.

---
eip: 5485
title: Legitimacy, Jurisdiction and Sovereignty
description: An interface for identifying the legitimacy, jurisdiction and sovereignty.
author: Zainan Victor Zhou (@xinbenlv)
discussions-to: https://ethereum-magicians.org/t/erc-5485-interface-for-legitimacy-jurisdiction-and-sovereignty/10425
status: Draft
type: Standards Track
category: ERC
created: 2022-08-17

> requires: 165

Reviewer: ```suggestion
requires: 165, 5247
```

Author: Good catch, thanks!

---

## Abstract
Provide a way for compliant smart contracts to declare their legitimacy lineage, jurisdiction they observe, and sovereignty if they choose to not fall onto any jurisdiction.

## Motivation

> Today, smart contract has no standard way to specify their legitimacy lineage, jurisdiction and sovereignty relationship.
> By supporting such a standard, it supports better integrate with today's legal and regulative scencario: 
> 1. it supports a regulative body to allow or deny interoperatbility with smart contracts. 
> 2. it also allows DAOs to clearely declare "self-sovereignty" by announcing via this interface by saying they do not assert legitimacy from any source
> other then themselves.

Reviewer: ```suggestion
Today, smart contracts have no standard way to specify their legitimacy lineage, jurisdiction, or sovereignty relationship. The introduction of such a standard, supports better integration with today's legal and regulative scenarios:

1. it supports a regulative body to allow or deny interoperability with smart contracts. 
2. it also allows DAOs to clearly declare "self-sovereignty" by announcing via this interface by saying they do not assert legitimacy from any source other than themselves.
```
Formatting fix required; spelling fixes required; grammatical fix required; rephrasing optional but recommended.

> A real world example is that ContractA represents an **A company registered in a country**,
> ContractB represents a **The Secretary of State of the country**. The ContractC represents **Supreme Court of the Country**.

Reviewer: ```suggestion
A real-world example is that ContractA represents an **A company registered in a country**, ContractB represents a **The Secretary of State of the country**, and ContractC represents the **Supreme Court of the Country**.
```
Formatting fix required.

> Another real example is a contract the declared "self-sovereignty" that doesn't follow any jurisdiction.

Reviewer: ```suggestion
Another real example is a contract that declares "self-sovereignty" that doesn't follow any jurisdiction.
```
Grammatical fix required.

> This interface supports both cases, providing a way to allow smart contract to determine if they want to allow / prohibit interaction based
> on the sovereignty.

Reviewer: ```suggestion
This interface supports both cases, providing a way to allow smart contracts to determine if they want to allow/prohibit interaction based on sovereignty.
```
Grammatical fix required; formatting fix required.

> For example A country might want to require any digital money service's all smart contracts to observe their [EIP-5485](./eip-5485.md) jurisdiction before they are allow to operate money in their (real world) legal jurisdiction.

Reviewer: ```suggestion
For example, a country might want to require any digital money service's all smart contracts to observe their [EIP-5485](./eip-5485.md) jurisdiction before they are allowed to operate money in their (real world) legal jurisdiction.
```
Grammatical fix required.

> On the other hand, a DAO with strong decentralized ideology might choose to only inter-operate with EOA
> or "self-sovereigned" smart contracts to avoid being affected by any countries.

Reviewer: ```suggestion
On the other hand, a DAO with a strong decentralized ideology might choose to only inter-operate with EOA or "self-sovereign" smart contracts to avoid being affected by any country.
```
Formatting fix required; grammatical fix(es) required.

## Specification
The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.

1. Compliant contract MUSTS implement the following interface.

```solidity
interface IERC5485 {
    /// A function that returns the source of legitimacy in an address
    ///
    /// @returns address of EOA or contract that grants legitimacy status to this contract. 
    /// See Specification #2 and #3 for more details.
    function sourceOfLegitimacy() public view returns (address);
    
    /// A function that returns the sovereign jurisdiction in an address
    ///
    /// @returns address of EOA or contract that *ultimately* grants legitimacy status (root of legitimacy lineage). 
    /// See Specification #4 and #4 for more details.

>     function jurisdiction() public view returns (address);

Reviewer: This isn't a blocker, but what about things that fall into multiple jurisdictions?

Author: That's a good question. I am thinking of that too. 
In United States model;
1. Prior to founding of the nation, The colony states are not self-sovereign. They gain legitimacy from UK.
2. States each gains sovereignty after the independence from UK,
3. States lost their individual sovereignty after American Civil War, with the north winning, the other rights in federal constitution take priority over state's right to leave the Union.

The European Union is still sovereign at the country level, especially with UK leaving EU, it signifies that the right to leave as a EU member takes priority on supporting other rights

Supporting multi-legitimacy / jurisdiction might complicate the model too much hence I am weak leaning towards a single jurisdiction to make everything simple. Leaving the challenge of multi-jurisdiction to future works

}
```

> 2. Suppose a compliant `ContractA` is deployed at address `0x<Contract_Address_A>` gains its legitimacy from another contract or External Owned Account (EOA) at `0x<ADDRESS_B>`, it shall return the `0x<ADDRESS_B>` when being queried for `sourceOfLegitimacy()`. For example. If `ContractA` gains legitimacy from `ContractB`, `ContractB` gains legitimacy from `ContractC` but `ContractC` doesn't gain legitimacy from any one, ContractC is considered the "jurisdiction" that both `ContractB` and `ContractA` observe.
> Both `ContractA` and `ContractB` MUST return address of `ContractC` as the result of jurisdiction.

Reviewer: ```suggestion
2. Suppose a compliant `ContractA` is deployed at address `0x<Contract_Address_A>` and gains its legitimacy from another contract or External Owned Account (EOA) at `0x<ADDRESS_B>`. In this case, it MUST return the `0x<ADDRESS_B>` when being queried for `sourceOfLegitimacy()`. For example. If `ContractA` gains legitimacy from `ContractB`, `ContractB` gains legitimacy from `ContractC` but `ContractC` doesn't gain legitimacy from anyone, ContractC is considered the "jurisdiction" that both `ContractB` and `ContractA` observe. Both `ContractA` and `ContractB` MUST return the address of `ContractC` as the result of jurisdiction.
```
Formatting fix required; grammatical fix required; typographical fix required; rephrasing recommended but optional.
